src/libs/ajax.js

```diff
@@ -0,0 +1,1012 @@
+    return await res.json();
+  },
+
+  findDictionary: async () => {
+    const url = `${await Config.getApiUrl()}/dataset/dictionary`;
+    const res = await fetchOk(url, Config.authOpts());
+    return await res.json();
+  },
+
+  findDataSets: async dacUserId => {
+    const url = `${await Config.getApiUrl()}/dataset?dacUserId=${dacUserId}`;
+    const res = await fetchOk(url, Config.authOpts());
+    return await res.json();
+  },
+
+  getDataSetsByDatasetId: async dataSetId => {
+    const url = `${await Config.getApiUrl()}/dataset/${dataSetId}`;
+    const res = await fetchOk(url, Config.authOpts());
+    return await res.json();
+  },
+
+  downloadDataSets: async (objectIdList, fileName) => {
+    const url = `${await Config.getApiUrl()}/dataset/download`;
+    const res = await fetchOk(url, _.mergeAll([Config.jsonBody(objectIdList), Config.fileOpts(), { method: 'POST' }]));
+
+    fileName = fileName === null ? getFileNameFromHttpResponse(res) : fileName;
+    const responseObj = await res.json();
+
+    let blob = new Blob([responseObj.datasets], { type: 'text/plain' });
+    const urlBlob = window.URL.createObjectURL(blob);
+    let a = document.createElement('a');
+    a.href = urlBlob;
+    a.download = fileName;
+    a.click();
+  },
+
+  deleteDataset: async (datasetObjectId, dacUserId) => {
+    const url = `${await Config.getApiUrl()}/dataset/${datasetObjectId}/${dacUserId}`;
+    const res = await fetchOk(url, _.mergeAll([Config.authOpts(), { method: 'DELETE' }]));
+    return await res;
+  },
+
+  disableDataset: async (datasetObjectId, active) => {
+    const url = `${await Config.getApiUrl()}/dataset/disable/${datasetObjectId}/${active}`;
+    const res = await fetchOk(url, _.mergeAll([Config.authOpts(), { method: 'DELETE' }]));
+    return res;
+  },
+
+  reviewDataSet: async (dataSetId, needsApproval) => {
+    const url = `${await Config.getApiUrl()}/dataset?dataSetId=${dataSetId}&needsApproval=${needsApproval}`;
+    const res = await fetchOk(url, _.mergeAll([Config.authOpts(), { method: 'PUT' }]));
+    return res.json();
+  }
+};
+
+export const DatasetAssociation = {
+
+  createDatasetAssociations: async (objectId, usersIdList) => {
+    const url = `${await Config.getApiUrl()}/datasetAssociation/${objectId}`;
+    const res = await fetchOk(url, _.mergeAll([Config.authOpts(), Config.jsonBody(usersIdList), { method: 'POST' }]));
+    return await res.json();
+  },
+
+  getAssociatedAndToAssociateUsers: async (objectId) => {
+    const url = `${await Config.getApiUrl()}/datasetAssociation/${objectId}`;
+    const res = await fetchOk(url, Config.authOpts());
+    return await res.json();
+  },
+
+  updateDatasetAssociations: async (objectId, usersIdList) => {
+    const url = `${await Config.getApiUrl()}/datasetAssociation/${objectId}`;
+    const res = await fetchOk(url, _.mergeAll([Config.authOpts(), Config.jsonBody(usersIdList), { method: 'PUT' }]));
+    return res.json();
+  }
+
+};
+
+export const Election = {
```

Comment:
This file is huge ... we should break it up and this is a good place to do so.

Reply:
We follow similar Saturn pattern for services. Maybe we let it to grow it too much. We'll split it up.